core/src/models/Destination.ts

```diff
@@ -167,6 +170,11 @@ export class Destination extends CommonModel<Destination> {
   @Column(DataType.ENUM(...SYNC_MODES))
   syncMode: DestinationSyncMode;
 
+  @AllowNull(false)
+  @Default("continual")
+  @Column(DataType.ENUM(...DELIVERY_MODES))
+  deliveryMode: DestinationDeliveryMode;
```

Comment:
There's probably a test to add that `foo` can't be used as a delivery mode.  That's probably going to need a `@BeforeSave` validation check. 

This code here is about setting the types, but it won't validate the data coming in... I think.  The test will let us know for sure!